app/services/PaymentAPIService.scala

```diff
@@ -0,0 +1,78 @@
+package services
+
+import java.io.IOException
+
+import play.api.libs.json.{JsObject, JsValue, Json}
+import play.api.libs.ws.{WSClient, WSResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object PaymentAPIService {
+  case class Email(value: String)
+  object Email {
+    def fromResponse(resp: WSResponse): Either[Throwable, Option[Email]] = {
+      if (resp.status == 200) {
+        Right((resp.json \ "email").validate[String].asOpt.map(Email.apply))
+      } else {
+        Left(new IOException(resp.toString))
+      }
+    }
+  }
+}
+
+class PaymentAPIService(wsClient: WSClient, paymentAPIUrl: String, paymentApiPayPalExecutePaymentPath: String) {
```

Comment:
Again, do we need 2 separate parameters for the url here?

Reply:
I can concatenate it before passing to the method, I would like to keep it separate to reduce the duplication in the configuration files. I added the path in the general config and the different URL in the envs config. In that way, the env config only has the bit that is different. Do you think this is a good idea?